### PR TITLE
Add `CLASS_OF`

### PR DESCRIPTION
**Pull request by wagenet**

### CODE REVIEW

**Comment by wagenet** [comment]
@chancancode should be an easy review.

File: src/ruby.rs [hunk 1]
```diff
@@ -560,6 +560,24 @@ extern {
     pub static rb_eMathDomainError: VALUE;
 
 
+    /// Returns the Ruby class of the object.
+    ///
+    /// * `obj` - a Ruby object
+    ///
+    /// Similar to [`rb_obj_class`] but may return a singleton.
```
Comment (chancancode) [inline src/ruby.rs]: singleton class

File: src/ruby.rs [hunk 2]
```diff
@@ -560,6 +560,24 @@ extern {
...
+    ///
+    /// # Safety
+    ///
+    /// * Undefined behavior if `obj` is not a `Class`.
```
Comment (chancancode) [inline src/ruby.rs]: This seems incorrect

Comment (wagenet) [reply src/ruby.rs]: Whoops.

**Pull request merged**